src/til/ut_til/EnvTests.cpp

```diff
@@ -38,12 +40,15 @@ class EnvTests
         actual.regenerate();
 
         VERIFY_ARE_EQUAL(expected.as_map().size(), actual.as_map().size());
-        for (const auto& [expectedKey, expectedValue] : expected.as_map())
+
+        const auto expectedEnd = expected.as_map().end();
+        auto expectedIt = expected.as_map().begin();
+        auto actualIt = actual.as_map().begin();
+
+        for (; expectedIt != expectedEnd; ++expectedIt, ++actualIt)

```

Comment:
Careful! The original check is order-independent. This check is not. It was explicitly written this way because the test was failing.

It's just test code, it's allowed to be inefficient.

Reply:
Did it use an `unordered_map` before? Right now it uses a `map` and the iteration order of the two instances should match 1:1.

I made this change because I found the old code very confusing while testing my changes. This doesn't work:
```cpp
const auto actualValue = actual.as_map()[expectedKey];
VERIFY_IS_TRUE(actual.as_map().find(expectedKey) != actual.as_map().end());
```
The first line ensures that the second line always succeeds. This code is also somewhat confusing to me:
```cpp
VERIFY_ARE_EQUAL(expectedValue, actual.as_map()[expectedKey]);
```
because it doesn't properly work in case either of the two values is an empty string.

In short, my intention was to remove the use of `operator[]` to increase my confidence that I didn't break anything.

Reply:
AH. Thanks.